test/jdk/java/awt/event/MouseEvent/MouseRButTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 1998, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4037521
+ * @summary Mouse Right button does not send mouseClick action
+ * @key headful
+ * @library /java/awt/regtesthelpers
+ * @build Util
+ * @run main MouseRButTest
+ */
+
+import java.awt.Button;
+import java.awt.EventQueue;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import test.java.awt.regtesthelpers.Util;
+
+public class MouseRButTest {
+    private static Frame frame;
+    private static Button button;
+    private static final CountDownLatch latch = new CountDownLatch(1);
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        try {
+            EventQueue.invokeAndWait(MouseRButTest::createAndShowGUI);
+
+            robot.waitForIdle();
+            robot.delay(500);
+
+            Point point = Util.invokeOnEDT(() -> button.getLocationOnScreen());
```

Comment:
Received a suggestion on one of my other PRs that awt/regtesthelper/Util can be replaced with swing/regtesthelper/Util.getCenterPoint(button) (makes it easier and it is already called on EDT)

Reply:
Yes, sometimes it may be helpful